src/services/importService.ts

```diff
@@ -0,0 +1,191 @@
+        convertedProject = {
+            id: shortid.generate(),
+            name: project.file.name.split(".")[0],
+            version: "v1-to-v2",
+            securityToken: `${project.file.name.split(".")[0]} Token`,
+            description: "Converted V1 Project",
+            tags: parsedTags,
+            sourceConnection: connections[0],
+            targetConnection: connections[1],
+            exportFormat: null,
+            videoSettings: {
+                frameExtractionRate: 15,
+            },
+            autoSave: true,
+            assets: {},
+        };
+
+        const assetService = new AssetService(convertedProject);
+        generatedAssetMetadata = this.generateAssets(project, assetService);
+
+        const saveAssets = generatedAssetMetadata.map((assetMetadata) => {
+            assetMetadata.then((metadata) => {
+                return assetService.save(metadata);
+            });
+        });
+
+        try {
+            Promise.all(saveAssets);
+        } catch (e) {
+            throw e;
+        }
+
+        return convertedProject;
+    }
+
+    /**
+     * Generate connections from v1 project file location
+     * @param project - V1 Project Content and File Information
+     */
+    private generateConnections(project: any): IConnection[] {
+        const sourceConnection: IConnection = {
+            id: shortid.generate(),
+            name: `${project.file.name.split(".")[0]} Source Connection`,
+            providerType: "localFileSystemProxy",
+            providerOptions: {
+                folderPath: project.file.path.replace(/[^\/]*$/, ""),
+            },
+        };
+
+        const targetConnection: IConnection = {
+            id: shortid.generate(),
+            name: `${project.file.name.split(".")[0]} Target Connection`,
+            providerType: "localFileSystemProxy",
+            providerOptions: {
+                folderPath: project.file.path.replace(/[^\/]*$/, ""),
+            },
+        };
+
+        const connections: IConnection[] = [sourceConnection, targetConnection];
+
+        return(connections);
+    }
+
+    /**
+     * Parse v1 project's tag string and return array of ITags
+     * @param project - V1 Project Content and File Information
+     */
+    private parseTags(project: any): ITag[] {
+        const finalTags: ITag[] = [];
+        const tagStrings = project.inputTags.split(",");
+        const tagColors = project.tag_colors;
+
+        for (let i = 0; i < tagColors.length; i++) {
+            const newTag = {
+                name: tagStrings[i],
+                color: tagColors[i],
+            };
+            finalTags.push(newTag);
+        }
+        return finalTags;
+    }
+
+    /**
+     * Generate assets based on V1 Project frames and regions
+     * @param project - V1 Project Content and File Information
+     */
+    private generateAssets(project: any, assetService: AssetService): Array<Promise<IAssetMetadata>> {
```

Comment:
I would update this method to return the final asset metadata for the assets vs. the promises.